src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStrip.cs

```diff
@@ -550,6 +551,11 @@ public override Font Font
             // thread local storage for it.
             _defaultFont ??= ToolStripManager.DefaultFont;
 
+            if (_defaultFont != base.Font && LocalAppContextSwitches.ApplyApplicationFontToMenus is true)
```

Comment:
```suggestion
            if (_defaultFont != base.Font && LocalAppContextSwitches.ApplyApplicationFontToMenus)
```

Also, consider switching to check the switch value first. If there is work being done in `base.Font` that we do not need unless the switch is `true` then we are doing extra work to no benefit.